hyperspy/roi.py

```diff
@@ -322,11 +322,13 @@ def interactive(self, signal, navigation_signal="same", out=None,
             If not None, it will use 'out' as the output instead of returning
             a new Signal.
         """
-        if navigation_signal == "same":
+        if isinstance(navigation_signal, basestring) and navigation_signal == "same":
             navigation_signal = signal
         if navigation_signal is not None:
             if navigation_signal not in self.signal_map:
                 self.add_widget(navigation_signal)
+        if self.update not in signal.events.data_changed.connected:
+            signal.events.data_changed.connect(self.update, [])
```

Comment:
Without this the 2D line in user-guide example with image -> rectangle roi -> 2d line roi does not get updated when modifying the position of rectangle roi - the line is obviously stationary in value space, but the data there does not exist / the line roi gets drawn out of bounds and is not limited any more by the rectangle axes. With this bit, it behaves as if the rectangle ROI is all of the data, and when moving it, the line (eventually, if moved too far) collapses into a point that is still reachable by the user.